kv/batch.go

```diff
@@ -40,6 +40,8 @@ func newRSpan(ba roachpb.BatchRequest) rSpan {
 
 // intersect returns the intersection of the current span and the
 // descriptor's range.
+// TODO(kkaneda): Currently the method returns desc when desc and the span
+// do not overlap. Fix if this is not the expected behavior here.
```

Comment:
probably best to return an error in that case. The use case in `DistSender` mandates that the intersection is nonempty, though a bug could possibly lead to that situation occurring.